sdk/cosmos/azure-cosmos-test/src/main/java/com/azure/cosmos/test/implementation/faultinjection/FaultInjectionRuleProcessor.java

```diff
@@ -351,10 +351,23 @@ private Mono<List<URI>> resolvePhysicalAddresses(
                                         addressEndpoints.isIncludePrimary(),
                                         true)
                                     .flatMapIterable(addresses -> {
+                                        // There are two rules need to happens here:
+                                        // 1. if isIncludePrimary is true, then basically make sure primary replica address will always be returned
+                                        // 2. make sure the same replica addresses will be used across different client instances
                                         return addresses
                                             .stream()
+                                            .sorted((o1, o2) -> {
```

Comment:
the sorting is only on primary or not, how will it make sure that ordering is maintained across all client instances?